code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -66,41 +65,33 @@ func (fsh *StorageHandler) convertGormError(err error) error {
 	return err
 }
 
-func (fsh *StorageHandler) verifyAuthTicket(ctx context.Context, authTokenString string, allocationObj *allocation.Allocation, refRequested *reference.Ref, clientID string) (bool, error) {
+// verifyAuthTicket verifies authTicket and returns authToken and error if any. For any error authToken is nil
+func (fsh *StorageHandler) verifyAuthTicket(ctx context.Context, authTokenString string, allocationObj *allocation.Allocation, refRequested *reference.Ref, clientID string) (*readmarker.AuthTicket, error) {
 	if authTokenString == "" {
-		return false, common.NewError("invalid_parameters", "Auth ticket required if data read by anyone other than owner.")
+		return nil, common.NewError("invalid_parameters", "Auth ticket is required")
 	}
+
 	authToken := &readmarker.AuthTicket{}
-	err := json.Unmarshal([]byte(authTokenString), &authToken)
-	if err != nil {
-		return false, common.NewError("invalid_parameters", "Error parsing the auth ticket for download."+err.Error())
+	if err := json.Unmarshal([]byte(authTokenString), &authToken); err != nil {
+		return nil, common.NewError("invalid_parameters", "Error parsing the auth ticket for download."+err.Error())
 	}
-	err = authToken.Verify(allocationObj, clientID)
-	if err != nil {
-		return false, err
+
+	if err := authToken.Verify(allocationObj, clientID); err != nil {
+		return nil, err
 	}
+
 	if refRequested.LookupHash != authToken.FilePathHash {
 		authTokenRef, err := reference.GetReferenceFromLookupHash(ctx, authToken.AllocationID, authToken.FilePathHash)
 		if err != nil {
-			return false, err
-		}
-
-		parentPath := refRequested.ParentPath
```

Comment:
@lpoli There was the bug, when I shared the root folder I was not able to access inner files/folders with authTicket. This code resolved it, not sure why it's commented now